app/controllers/bookmarks_controller.rb

```diff
@@ -32,6 +32,21 @@ def edit
 	  authorize @bookmark
 	end
 
+  def update
+    @bookmark = Bookmark.find(params[:id])
+    @topic = Topic.find(params[:topic_id])
+    @user = Bookmark.find(params[:user_id])
+    authorize @bookmark
+
+    if @bookmark.update_attributes(bookmark_params)
+      flash[:notice] = "Your bookmark has been sucessfully updated"
+      redirect_to topic_path(@topic.id)
+    else
+      flash[:error] = "Error updating, please try again"
+      render :edit
+    end
+  end
+
 	def destroy
     @bookmark = Bookmark.find(params[:id])
```

Comment:
Apply previous feedback on this PR to fetch `@bookmark` and `@topic` more securely.